pyomo/contrib/interior_point/inverse_reduced_hessian.py

```diff
@@ -0,0 +1,124 @@
+import pyomo.environ as pyo
+from pyomo.opt import check_optimal_termination
+from pyomo.common.dependencies import attempt_import
+from .interface import InteriorPointInterface
+from .linalg.scipy_interface import ScipyInterface
+
```

Comment:
I know that I am responsible for this, but should we move this file somewhere else?
Also, does it work with this interior-point method, or does it truly need IPOPT?

Reply:
I'm not sure where we would move it to? Do we need a directory for this one file? I vote we move it later. I'm sure it will be more clear where it should live after this module matures a bit.

I think it should work with this interior-point method if it converges. The algorithm has an inertia correction now, so if it converges, it should be correct.

Reply:
Ya, I think it is the reliance on Ipopt suffixes that is more problematic.